cpp/test/neighbors/ann_cagra.cuh

```diff
@@ -214,6 +295,79 @@ class AnnCagraTest : public ::testing::TestWithParam<AnnCagraInputs> {
   rmm::device_uvector<DataT> search_queries;
 };
 
+template <typename DistanceT, typename DataT, typename IdxT>
+class AnnCagraSortTest : public ::testing::TestWithParam<AnnCagraInputs> {
+ public:
+  AnnCagraSortTest()
+    : stream_(handle_.get_stream()),
+      ps(::testing::TestWithParam<AnnCagraInputs>::GetParam()),
+      database(0, stream_)
+  {
+  }
+
+ protected:
+  void testCagraSort()
+  {
+    {
+      // Step 1: Build a sorted KNN graph by CAGRA knn build
+      auto database_view = raft::make_device_matrix_view<const DataT, IdxT>(
+        (const DataT*)database.data(), ps.n_rows, ps.dim);
+      auto database_host = raft::make_host_matrix<DataT, IdxT>(ps.n_rows, ps.dim);
+      raft::copy(database_host.data_handle(), database.data(), database.size(), stream_);
+      auto database_host_view = raft::make_host_matrix_view<const DataT, IdxT>(
+        (const DataT*)database_host.data_handle(), ps.n_rows, ps.dim);
+
+      cagra::index_params index_params;
+      auto knn_graph =
+        raft::make_host_matrix<IdxT, IdxT>(ps.n_rows, index_params.intermediate_graph_degree);
+
+      if (ps.host_dataset) {
+        cagra::build_knn_graph<DataT, IdxT>(handle_, database_host_view, knn_graph.view());
+      } else {
+        cagra::build_knn_graph<DataT, IdxT>(handle_, database_view, knn_graph.view());
+      };
+
+      handle_.sync_stream(stream_);
+      handle_.sync_stream();
```

Comment:
Since` stream_ == handle.get_stream()`, this two statements are actually equivalent. (It is somewhat obscure but the we just [call this](https://github.com/rapidsai/raft/blob/a44ca96c5cddec7ca67b510f3c21163d3958bc7e/cpp/include/raft/core/resource/cuda_stream.hpp#L100) in the second case.
```suggestion
      handle_.sync_stream();
```